village/src/App.js

```diff
@@ -11,17 +17,57 @@ class App extends Component {
       smurfs: [],
     };
   }
+
+  componentDidMount(){
+    axios.get('http://localhost:3333/smurfs')
+    .then(res=>{
+      console.log(res);
+      this.setState({smurfs:res.data})})
+      .catch(err=>{
+        console.log(err)
+      })
+    }
+
+    addSmurf=smurf=>{
+      axios.post('http://localhost:3333/smurfs',smurf)
+      .then(res=>{
+        this.setState({smurfs:res.data});
+        this.props.history.push('/smurfs');
```

Comment:
There's no `/smurfs` route, this should just be `/`
And this is in App.js, which does not have access to the Router props unless you use `withRouter`